Sources/APIClient/URLSession+Async.swift

```diff
@@ -0,0 +1,37 @@
+import Foundation
+
+extension URLSession {
+    @available(macOS, obsoleted: 12, message: "Use new URLSession data(for:) method instead")
+    @available(iOS, obsoleted: 15, message: "Use new URLSession data(for:) method instead")
+    @available(macCatalyst, obsoleted: 15, message: "Use new URLSession data(for:) method instead")
+    @available(watchOS, obsoleted: 8, message: "Use new URLSession data(for:) method instead")
+    @available(tvOS, obsoleted: 15, message: "Use new URLSession data(for:) method instead")
+    func asyncData(for request: URLRequest) async throws -> (Data, URLResponse) {
+        let cancellableTask = CancellableTask()
+        return try await withTaskCancellationHandler {
+            try Task.checkCancellation()
```

Comment:
There is probably no reason to check for cancellation this early.

Reply:
You're right. I was thinking about some "delayed execution" with `async let`, however, couldn't invent a test case for it.